webapp/packages/core-connections/src/Administration/Connections/ConnectionForm/Options/OptionsController.ts

```diff
@@ -104,36 +111,55 @@ implements IInitializableController {
       this.model.connection.url = this.driver?.sampleURL;
     }
 
-    this.updateName();
+    this.prevConnectionParameters.host = this.model.connection.host;
+    this.prevConnectionParameters.port = this.model.connection.port;
+    this.prevConnectionParameters.url = this.model.connection.url;
+
+    if (!this.model.editing) {
+      if (this.model.connection.url && !this.model.connection.host && !this.model.connection.port) {
+        this.model.connection.name = this.model.connection.url;
+      } else {
+        this.model.connection.name = this.getNameTemplate(this.model.connection.host, this.model.connection.port);
+      }
+    }
   }
 
   private updateName(name?: string) {
     if (name === 'name') {
       return;
     }
 
-    const databaseNames = ['New', ...this.drivers.map(driver => driver.name!)]
-      .filter(Boolean);
+    const host = this.model.connection.host;
+    const port = this.model.connection.port;
+    const url = this.model.connection.url || '';
+    const prevHost = this.prevConnectionParameters.host;
+    const prevPort = this.prevConnectionParameters.port;
+    const prevUrl = this.prevConnectionParameters.url;
 
-    const matches = this.nameTemplate.exec(this.model.connection.name);
+    if (url && !host && !port && this.model.connection.name === prevUrl) {
+      this.model.connection.name = url;
+      this.prevConnectionParameters.url = url;
+      return;
+    }
 
-    if (this.model.connection.name === undefined
-        || (matches?.length && databaseNames.includes(matches[1]))) {
-      this.model.connection.name = this.getNameTemplate();
+    if (this.model.connection.name === this.getNameTemplate(prevHost, prevPort)) {
+      this.model.connection.name = this.getNameTemplate(host, port);
+      this.prevConnectionParameters.host = host;
+      this.prevConnectionParameters.port = port;
     }
   }
 
-  private getNameTemplate() {
+  private getNameTemplate(host?: string, port?: string) {
```

Comment:
you should completely rewrite this function behavior:
1. function accepts current config
2. function check if current name equal to previously generated name by this function (if it's the first time, we decide that name is equal to the previously generated (null))
  false) return;
3. generate name based on config
4. save newly generated name to controller variable
5. update name in config

seems like you can mutate config object in this function, so usage will be:
`this.updateNameTemplate(config)`

also, you be able to use this function in `setDefaultParameters`